### PR TITLE
Added markdown-file import

### PR DESCRIPTION
**Pull request by ErikMichelson**
### Component/Part
Editor -> Document-Bar -> Import -> Markdown file

### Description
This PR adds support for importing markdown-files by appending their text content to the current note content.

### Steps

- [x] added implementation
- [x] added / updated tests
- [ ] added / updated documentation
- [ ] extended changelog

### Related Issue(s)
#622

### CODE REVIEW

**Comment by DerMolly** [comment]
As we use for the export of markdown files just the text `Markdown` maybe this would suffice for the import, too?

**Comment by ErikMichelson** [comment]
I explicitly decided for "Markdown file".
In the export context it's quite clear what the Markdown button does especially because of the heading _download_ above it.
As HedgeDoc primarily is a markdown note editor, the context "Import markdown" isn't quite intuitive. The label "Import markdown file" on the other side is more specific.

File: src/components/editor/document-bar/import/import-file.tsx [hunk 1]
```diff
@@ -17,19 +17,26 @@ export const ImportFile: React.FC<ImportProps> = ({ noteContent, updateNoteConte
...
       const fileReader = new FileReader()
-      fileReader.readAsText(file)
       fileReader.addEventListener('load', () => {
         const newContent = fileReader.result as string
-        updateNoteContent(noteContent + '\n' + newContent)
+        if (noteContent.length === 0) {
+          updateNoteContent(newContent)
+        } else {
+          updateNoteContent(noteContent + '\n' + newContent)
+        }
```
Comment (DerMolly) [inline src/components/editor/document-bar/import/import-file.tsx:26]: should an upload really be appended? I thought it should replace the current content and create a new revision.

Comment (ErikMichelson) [reply src/components/editor/document-bar/import/import-file.tsx:26]: I would differentiate between the API call `PUT /note ` and the editor UI. Afaik the other imports don't replace the whole note content either.

**Review by mrdrogdrog — APPROVED** [review]
Maybe we should think about moving the editor content into the redux state..

**Comment by ErikMichelson** [comment]
That could be an idea, but that's out of scope for this PR. Do you want to raise an issue for that?

**Comment by mrdrogdrog** [comment]
I didn't want you to solve this in this PR. That's my I approved it :yum:

**Review by DerMolly — APPROVED** [review]

**Pull request merged**